File: Chapter01/C#/Exercises/Ducks/MallardDuck.cs
namespace Exercises.Ducks
{
    public class MallardDuck : Duck
    {
        public MallardDuck()
        {
            quackBehavior = new Quack();
            flyBehavior = new FlyWithWings();
        }

        public override void Display()
        {
            System.Console.WriteLine("I'm a real mallard duck");
        }
    }
}

File: head-first-design-patterns/Chapter01/Java/Exercises/src/ducks/MallardDuck.java
package ducks;

public class MallardDuck extends Duck {

    public MallardDuck() {
        super();

        quackBehavior = new Quack();
        flyBehavior = new FlyWithWings();
    }

    @Override
    public void display() {
        System.out.println("I'm a real mallard duck");
    }

}